diff --git a/dmidecode.c b/dmidecode.c
index 83e66e6..a5304a7 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -2301,10 +2301,11 @@ static const char *dmi_memory_device_type(u8 code)
 		"Reserved",
 		"Reserved",
 		"DDR3",
-		"FBD2", /* 0x19 */
+		"FBD2",
+		"DDR4" /* 0x1A */
 	};
 
-	if (code >= 0x01 && code <= 0x19)
+	if (code >= 0x01 && code <= 0x1A)
 		return type[code - 0x01];
 	return out_of_spec;
 }
